lib/rucio/core/message.py

```diff
@@ -123,7 +125,16 @@ def retrieve_messages(bulk: int = 1000,
             Message.created_at
         )
         stmt_subquery = filter_thread_work(session=session, query=stmt_subquery, total_threads=total_threads, thread_id=thread)
-        if event_type:
+        if service_filter and event_type:
+            stmt_subquery = stmt_subquery.where(
+                Message.services == service_filter,
+                Message.event_type == event_type
+            )
+        elif service_filter:
+            stmt_subquery = stmt_subquery.where(
+                Message.services == service_filter
+            )
+        elif event_type:
             stmt_subquery = stmt_subquery.where(
                 Message.event_type == event_type
             )
```

Comment:
Can be simplified to:
```suggestion
        if service_filter:
            stmt_subquery = stmt_subquery.where(
                Message.services == service_filter
            )
        if event_type:
            stmt_subquery = stmt_subquery.where(
                Message.event_type == event_type
            )
```